### PR TITLE
ZNC-22: DOC: Add developer bootstrap guide

### PR DESCRIPTION
**Pull request by LaureVergeron**
This guide needs to be split into smaller files, but I thought it might be easier to review as one big files - experts, let me know.

### CODE REVIEW

File: docs/modules/DEVELOPERS/GETTING_STARTED.adoc [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+If that is something you want to contribute (or just do on your own version of
+the cloudserver image), go read our link:NEW_BACKEND.adoc[step-by-step guide] on
+where to start to add support for a new backend.
+//TODO:add link to contributing guidelines
+If you wish to make this a contribution, please make sure you follow our
+Contributing Guidelines.
+
+If you need help with anything, please search our https://forum.scality.com[Forum]
+for more information. If you can't find what you need, open a thread, and our
+community memebers and core team will be right with you!
```
Comment (alexanderchan-scality) [inline docs/modules/DEVELOPERS/GETTING_STARTED.adoc]: typo: "memebers" -> "members"

File: docs/modules/DEVELOPERS/GETTING_STARTED.adoc [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+== [[telling-story-usecase]]
+== Telling the community about your story or usecase
```
Comment (alexanderchan-scality) [inline docs/modules/DEVELOPERS/GETTING_STARTED.adoc]: use case, 2 words?

File: docs/modules/DEVELOPERS/GETTING_STARTED.adoc [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+The best part of being open source is learning from such a diverse crowd. At
+Scality, we're always curious to learn about what you do with Zenko and Zenko
+Cloudserver.
+If you wish to tell us your story, if you want us to advertise your extension,
+or if you want to publish a tutorial on how to replicatie your setup, please
```
Comment (alexanderchan-scality) [inline docs/modules/DEVELOPERS/GETTING_STARTED.adoc]: type "replicatie" -> "replicate"

File: docs/modules/DEVELOPERS/NEW_BACKEND.adoc [hunk 4]
```diff
@@ -0,0 +1,54 @@
+= Adding a new backend
+
+One of Zenko's Cloudserver commitment is to simplify multicloud storage by
+giving one API (the S3 API) to access all clouds. With that in mind, supporting
+more and more backends is one of Zenko's Community priorities. And you, as a
+developper, are welcome to join that trend!
```
Comment (alexanderchan-scality) [inline docs/modules/DEVELOPERS/NEW_BACKEND.adoc]: type "developper" -> "developer"

File: docs/modules/DEVELOPERS/NEW_BACKEND.adoc [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+course!);
+- when you think it's ready, let us know so that we create a feature branch
+against which we'll compare and review your code;
+- open a pull request with your changes against that dedicated feature branch;
+//TODO: Add Hall of Fame section in the community report
+- once that pull request gets merged, you're done (and you'll join our Hall of
+Fame ;) );
+- finally, we'll let you know when we merge this into master.
+
+TIP: While we do take care of the finale rebase (when we merge your feature
```
Comment (alexanderchan-scality) [inline docs/modules/DEVELOPERS/NEW_BACKEND.adoc]: "finale" -> "final"

File: docs/modules/DEVELOPERS/NEW_BACKEND.adoc [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+           _If your code is clean and your extension works nicely, we will be_
+           _glad to advertise it as part of the Zenko Galaxy_
+
+//TODO: Get approval for Zenko Galaxy as the name of our hub - sound appropriate with Orbit ;)
+
+There are two main types of backend you could want Zenko to support:
+
+== link:S3_COMPATIBLE_BACKENDS.adoc[S3 compatible data backends]
+
+== link:NON_S3_COMPATIBLE_BACKENDS.adoc[Data backends using another protocol
```
Comment (alexanderchan-scality) [inline docs/modules/DEVELOPERS/NEW_BACKEND.adoc]: link displayed incorrectly

File: docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc [hunk 7]
```diff
@@ -0,0 +1,469 @@
...
+https://forum.zenko.io[Zenko forum].
+
+== General configuration
+
+There are a number of constants and environment variables to define to support a
+new data backends; here is a list and where to find them:
+
+=== `/constants.js`
+
+- give you backend type a name, as part of the `externalBackends` object;
```
Comment (alexanderchan-scality) [inline docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc]: "you" -> "your"

File: docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc [hunk 8]
```diff
@@ -0,0 +1,469 @@
...
+=== `/lib/data/external/{{BackendName}}Client.js`
+
+- the function that is going to call your `put()` function is also called
+  `put()`, and it's defined in `/lib/data/multipleBackendGateway.js`;
+- define a function with signature like
+  `put(stream, size, keyContext, reqUids, callback)`; this is worth exploring a
+  bit more as these parameters are the same for all backends:
+//TODO: generate this from jsdoc
+-- `stream`: the stream of data you want to put in the cloud; if you're
+   unfamiliar with node.js strams, we suggest you start training, as we use them
```
Comment (alexanderchan-scality) [inline docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc]: "strams" -> "streams"

File: docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc [hunk 9]
```diff
@@ -0,0 +1,469 @@
...
+
+=== `/lib/data/external/{{BackendName}}Client.js`
+
+- the function that is going to call your `get()` function is also called
+  `get()`, and it's defined in `/lib/data/multipleBackendGateway.js`;
+- define a function with signature like
+  `get(objectGetInfo, range, reqUids, callback)`; this is worth exploring a
+  bit more as these parameters are the same for all backends:
+//TODO: generate this from jsdoc
+-- `objectGetInfo`: a dictionnary with two entries: `key`, the object key in the
```
Comment (alexanderchan-scality) [inline docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc]: "dictionnary" -> "dictionary"

File: docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc [hunk 10]
```diff
@@ -0,0 +1,469 @@
...
+=== `/lib/data/external/{{backendName}}_lib/`
+
+- this is where you should put all utility functions for your MPU operations,
+  and then import then in `/lib/data/external/{{BackendName}}Client.js`, to keep
+  your code clean;
+
+=== `lib/data/multipleBackendGateway.js`
+
+- edit the `createLOY` function to add your location type, and call your
+  `©reateMPU()`; see pseudocode below for a sample implementation, provided your
```
Comment (alexanderchan-scality) [inline docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc]: is this intended `createLOY` + special character `©reateMPU` ?

File: docs/modules/USERS/pages/ARCHITECTURE.adoc [hunk 11]
```diff
@@ -0,0 +1,979 @@
...
+Versioning]
+* http://docs.aws.amazon.com/AmazonS3/latest/dev/ObjectVersioning.html[Object
+Versioning]
+
+This document assumes familiarity with the details of Bucket Versioning,
+including null versions and delete markers, described in the above
+links.
+
+Implementation of Bucket Versioning in Zenko CloudServer
+-----------------------------------------
```
Comment (alexanderchan-scality) [inline docs/modules/USERS/pages/ARCHITECTURE.adoc]: dashed line is causing asciidoc to generate incorrectly

File: docs/modules/DEVELOPERS/GETTING_STARTED.adoc (hunk 1 above)
Comment (LaureVergeron) [reply docs/modules/DEVELOPERS/GETTING_STARTED.adoc]: thanks

File: docs/modules/DEVELOPERS/GETTING_STARTED.adoc (hunk 2 above)
Comment (LaureVergeron) [reply docs/modules/DEVELOPERS/GETTING_STARTED.adoc]: Fixed, thanks

File: docs/modules/DEVELOPERS/GETTING_STARTED.adoc (hunk 3 above)
Comment (LaureVergeron) [reply docs/modules/DEVELOPERS/GETTING_STARTED.adoc]: fixed

File: docs/modules/DEVELOPERS/NEW_BACKEND.adoc (hunk 4 above)
Comment (LaureVergeron) [reply docs/modules/DEVELOPERS/NEW_BACKEND.adoc]: fixed

File: docs/modules/DEVELOPERS/NEW_BACKEND.adoc (hunk 5 above)
Comment (LaureVergeron) [reply docs/modules/DEVELOPERS/NEW_BACKEND.adoc]: fixed

File: docs/modules/DEVELOPERS/NEW_BACKEND.adoc (hunk 6 above)
Comment (LaureVergeron) [reply docs/modules/DEVELOPERS/NEW_BACKEND.adoc]: thanks

File: docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc (hunk 7 above)
Comment (LaureVergeron) [reply docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc]: fixed

File: docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc (hunk 8 above)
Comment (LaureVergeron) [reply docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc]: fixed

File: docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc (hunk 9 above)
Comment (LaureVergeron) [reply docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc]: fixed all, thanks

File: docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc (hunk 10 above)
Comment (LaureVergeron) [reply docs/modules/DEVELOPERS/NON_S3_COMPATIBLE_BACKENDS.adoc]: Absolutely not, good catch

File: docs/modules/USERS/pages/ARCHITECTURE.adoc (hunk 11 above)
Comment (LaureVergeron) [reply docs/modules/USERS/pages/ARCHITECTURE.adoc]: Yes, dropping all these for now, Antora will have its own PR. Thanks though!

**Comment by LaureVergeron** [comment]
@alexanderchan-scality : thanks for the review! I adressed all typos.
If you have time, what I'm most curious about is whether you think I'm pointing people to the right places in the code, and whether the explanations provided are sufficient or not? Thanks again!

File: docs/developers/GETTING_STARTED.adoc [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+:Email: <zenko@scality.com>
+
+[.lead]
+This set of documents aims at bootstrapping developpers with Zenko's Cloudserver
+module, so they can then go on and contribute features.
+In order to achieve this, we're going to cover a number of subjects:
+
+- <<cloning-and-building,cloning, installing, and building your own image>>; 
+- <<support-new-public-cloud, adding support to a new public cloud backend>>;
+- <<telling-story-usecase, telling the community about your story or usecase>>.
```
Comment (anurag4DSB) [inline docs/developers/GETTING_STARTED.adoc]: telling-story-usecase -> looks like a link. Is it intended to be a link?

File: docs/developers/NEW_BACKEND.adoc [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+developer, are welcome to join that trend!
+
+If you're planning to add a new backend for your own usage, go ahead and read
+the doc. If you have any questions during the development process, search our
+https://forum.scality.com[forum] and, if there is no answer to your question
+already there, open a new thread.
+
+//TODO: Add link to contributing Guidelines
+If you're planning to contribute your backend support to our official
+repository, please follow these steps:
```
Comment (anurag4DSB) [inline docs/developers/NEW_BACKEND.adoc]: Need a new line after this line to make sure the bullets are created. Currently, everything below this line is a big paragraph.

**Comment by smaffulli** [comment]
I think this has reached a 'good enough to get started' level and we should merge asap. Any objection?

File: docs/developers/GETTING_STARTED.adoc [hunk 14]
```diff
@@ -0,0 +1,91 @@
+Zenko Cloudserver for Developpers
+=================================
+:Revision: v1.0
+:Date: 2018-03-20
+:Email: <zenko@scality.com>
+
+[.lead]
+This set of documents aims at bootstrapping developpers with Zenko's Cloudserver
```
Comment (JianqinWang) [inline docs/developers/GETTING_STARTED.adoc]: 'developers'

File: docs/developers/GETTING_STARTED.adoc [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+ ~# cd cloudserver
+
+To install all dependencies (necessary to run), do:
+
+ ~/cloudserver# npm install
+
+TIP: Some optional dependencies may fail, resulting in you seeing `NPM WARN`
+     messages; these can safely be ignored.
+
+// Add link to user doc
```
Comment (JianqinWang) [inline docs/developers/GETTING_STARTED.adoc]: Is this a TODO comment?

File: docs/developers/GETTING_STARTED.adoc [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+To run the service locally, use:
+
+ ~/cloudserver# npm start
+
+TIP: Refer to the User documentation for all available options
+
+// Add link to Docker doc
```
Comment (JianqinWang) [inline docs/developers/GETTING_STARTED.adoc]: TODO comment?

File: docs/developers/GETTING_STARTED.adoc [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+|Backend type    |Currently supported |Active WIP |Community suggestion 
+|Private disk/fs |x                   |           |                     
+|AWS S3          |x                   |           |                     
+|Microsoft Azure |x                   |           |                     
+|Backblaze B2    |                    |x          |                     
+|Google Cloud    |                    |x          |                     
+|Openstack Swift |                    |           |x                    
+|=======================================================================
+
+IMPORTANT: Should you want to request a new backend support, please do so by
```
Comment (JianqinWang) [inline docs/developers/GETTING_STARTED.adoc]: Reads a little awkwardly, maybe something like "Should you want to request for a new backend to be supported, ..."

File: docs/developers/GETTING_STARTED.adoc [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+           opening a Github issue, and filling out the "Feature Request" section
+           of our template. Thanks!
+
+We always encourage our community to offer new extensions to Zenko, and new
+backend support is paramount to meeting more community needs.
+If that is something you want to contribute (or just do on your own version of
+the cloudserver image), go read our link:NEW_BACKEND.adoc[step-by-step guide] on
+where to start to add support for a new backend.
+//TODO:add link to contributing guidelines
```
Comment (JianqinWang) [inline docs/developers/GETTING_STARTED.adoc]: another TODO

File: docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc [hunk 19]
```diff
@@ -0,0 +1,469 @@
...
+
+These backends are what makes Zenko so valuable: abstracting the complexity of
+multiple APIs to let users work on a single common namespace across multiple
+clouds.
+
+This documents aims at introducing you to the right files in Cloudserver (the
+Zenko stack's subcomponent in charge of API translation, among other things) to
+add support to your own backend of choice.
+
+As usual, should you have any question, please reach out on the
```
Comment (JianqinWang) [inline docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc]: 'any questions'

File: docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc [hunk 20]
```diff
@@ -0,0 +1,469 @@
...
+https://forum.zenko.io[Zenko forum].
+
+== General configuration
+
+There are a number of constants and environment variables to define to support a
+new data backends; here is a list and where to find them:
```
Comment (JianqinWang) [inline docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc]: new data backends

File: docs/developers/NEW_BACKEND.adoc [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+it over time);
+- once approved, fork this https://www.github.com/scality/S3[repository], and
+get started!
+- reach out to us on the https://forum.scality.com[forum] with any question you
+may have during the development process (after reading this document, of
+course!);
+- when you think it's ready, let us know so that we create a feature branch
+against which we'll compare and review your code;
+- open a pull request with your changes against that dedicated feature branch;
+//TODO: Add Hall of Fame section in the community report
```
Comment (JianqinWang) [inline docs/developers/NEW_BACKEND.adoc]: TODO.

File: docs/developers/NEW_BACKEND.adoc [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+     then; find out more https://help.github.com/articles/syncing-a-fork/[here].
+
+IMPORTANT: If we do not approve your feature request, you may of course still
+           work on supporting a new backend: all our "no" means is that we do
+           not have the resources, as part of our core development team, to
+           maintain this feature for the moment.
+           _If your code is clean and your extension works nicely, we will be_
+           _glad to advertise it as part of the Zenko Galaxy_
+
+//TODO: Get approval for Zenko Galaxy as the name of our hub - sound appropriate with Orbit ;)
```
Comment (JianqinWang) [inline docs/developers/NEW_BACKEND.adoc]: TODO

File: docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc [hunk 23]
```diff
@@ -0,0 +1,469 @@
...
+These are the files you'll need to edit:
+
+=== `/lib/data/external/{{BackendName}}Client.js`
+
+- the function that is going to call your `put()` function is also called
+  `put()`, and it's defined in `/lib/data/multipleBackendGateway.js`;
+- define a function with signature like
+  `put(stream, size, keyContext, reqUids, callback)`; this is worth exploring a
+  bit more as these parameters are the same for all backends:
+//TODO: generate this from jsdoc
```
Comment (JianqinWang) [inline docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc]: TODO

File: docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc [hunk 24]
```diff
@@ -0,0 +1,469 @@
...
+These are the files you'll need to edit:
+
+=== `/lib/data/external/{{BackendName}}Client.js`
+
+- the function that is going to call your `get()` function is also called
+  `get()`, and it's defined in `/lib/data/multipleBackendGateway.js`;
+- define a function with signature like
+  `get(objectGetInfo, range, reqUids, callback)`; this is worth exploring a
+  bit more as these parameters are the same for all backends:
+//TODO: generate this from jsdoc
```
Comment (JianqinWang) [inline docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc]: TODO

File: docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc [hunk 25]
```diff
@@ -0,0 +1,469 @@
...
+These are the files you'll need to edit:
+
+=== `/lib/data/external/{{BackendName}}Client.js`
+
+- the function that is going to call your `delete()` function is also called
+  `delete()`, and it's defined in `/lib/data/multipleBackendGateway.js`;
+- define a function with signature like
+  `delete(objectGetInfo, reqUids, callback)`; this is worth exploring a
+  bit more as these parameters are the same for all backends:
+//TODO: generate this from jsdoc
```
Comment (JianqinWang) [inline docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc]: todo

File: docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc [hunk 26]
```diff
@@ -0,0 +1,469 @@
...
+These are the files you'll need to edit:
+
+=== `/lib/data/external/{{BackendName}}Client.js`
+
+- the function that is going to call your `head()` function is also called
+  `head()`, and it's defined in `/lib/data/multipleBackendGateway.js`;
+- define a function with signature like
+  `head(objectGetInfo, reqUids, callback)`; this is worth exploring a
+  bit more as these parameters are the same for all backends:
+//TODO: generate this from jsdoc
```
Comment (JianqinWang) [inline docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc]: TODO

File: docs/developers/NEW_BACKEND.adoc [hunk 27]
```diff
@@ -0,0 +1,53 @@
+= Adding a new backend
+
+One of Zenko's Cloudserver commitment is to simplify multicloud storage by
+giving one API (the S3 API) to access all clouds. With that in mind, supporting
```
Comment (JianqinWang) [inline docs/developers/NEW_BACKEND.adoc]: 'using' instead of 'giving'

File: docs/developers/NEW_BACKEND.adoc [hunk 28]
```diff
@@ -0,0 +1,53 @@
...
+more and more backends is one of Zenko's Community priorities. And you, as a
+developer, are welcome to join that trend!
+
+If you're planning to add a new backend for your own usage, go ahead and read
+the doc. If you have any questions during the development process, search our
+https://forum.scality.com[forum] and, if there is no answer to your question
+already there, open a new thread.
+
+//TODO: Add link to contributing Guidelines
```
Comment (JianqinWang) [inline docs/developers/NEW_BACKEND.adoc]: TODO

File: docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc [hunk 29]
```diff
@@ -0,0 +1,469 @@
...
+
+=== [[location-config-test-json]]
+=== `/tests/locationConfig/locationConfigTests.json`
+
+- this file is where you'll create location profiles to be used by your
+  functional tests;
+
+=== `/lib/data/locationConstraintParser.js`
+
+- this is where you'll instantiate your client if the operation the end user
```
Comment (JianqinWang) [inline docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc]: This sentence is phrased awkwardly, and I'm having trouble understanding the original intent.

File: docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc [hunk 30]
```diff
@@ -0,0 +1,469 @@
...
+-- `reqUids`: the request unique ID used for logging;
+-- `callback`: your function's callback (should handle errors);
+
+=== `/lib/data/external/{{backendName}}_lib/`
+
+- this is where you should put all utility functions for your DELETE operation,
+  and then import then in `/lib/data/external/{{BackendName}}Client.js`, to keep
+  your code clean;
+
+=== `tests/functional/aws-node-sdk/test/multipleBackend/get/get{{BackendName}}js`
```
Comment (JianqinWang) [inline docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc]: URL should be '/delete/delete{{BackendName}}.js'

File: docs/developers/NEW_BACKEND.adoc (hunk 13 above)
Comment (LaureVergeron) [reply docs/developers/NEW_BACKEND.adoc]: fixed :)

File: docs/developers/GETTING_STARTED.adoc (hunk 14 above)
Comment (LaureVergeron) [reply docs/developers/GETTING_STARTED.adoc]: fixed, thanks

File: docs/developers/GETTING_STARTED.adoc (hunk 15 above)
Comment (LaureVergeron) [reply docs/developers/GETTING_STARTED.adoc]: not up to date anymore, removed, thanks

File: docs/developers/GETTING_STARTED.adoc (hunk 16 above)
Comment (LaureVergeron) [reply docs/developers/GETTING_STARTED.adoc]: not up to date anymore, removed, thanks

File: docs/developers/GETTING_STARTED.adoc (hunk 17 above)
Comment (LaureVergeron) [reply docs/developers/GETTING_STARTED.adoc]: Fixed, thanks

File: docs/developers/GETTING_STARTED.adoc (hunk 18 above)
Comment (LaureVergeron) [reply docs/developers/GETTING_STARTED.adoc]: updated, thanks

File: docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc (hunk 20 above)
Comment (LaureVergeron) [reply docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc]: fixed

File: docs/developers/NEW_BACKEND.adoc (hunk 21 above)
Comment (LaureVergeron) [reply docs/developers/NEW_BACKEND.adoc]: Yes, but that is for later, Hall of Fame doesn't exist currently

File: docs/developers/NEW_BACKEND.adoc (hunk 22 above)
Comment (LaureVergeron) [reply docs/developers/NEW_BACKEND.adoc]: @smaffulli : should we just delete this?

File: docs/developers/NEW_BACKEND.adoc (hunk 27 above)
Comment (LaureVergeron) [reply docs/developers/NEW_BACKEND.adoc]: fixed thanks

File: docs/developers/NEW_BACKEND.adoc (hunk 28 above)
Comment (LaureVergeron) [reply docs/developers/NEW_BACKEND.adoc]: fixed

File: docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc (hunk 29 above)
Comment (LaureVergeron) [reply docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc]: Trying to say that this is where the new data backend client will be instantiated

File: docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc (hunk 30 above)
Comment (LaureVergeron) [reply docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc]: fixed thanks

**Comment by LaureVergeron** [comment]
@JianqinWang : thanks for the review; all leftover todos are for later stages, when we have a path to generate doc from the existing jsdoc/when we start using swagger/when the community grows. Currently, there is no quick way to make this happen, I believe, so this might stay like it is for this first draft. Let me know what you think.

**Comment by rahulreddy** [comment]
Closing this in favor of https://github.com/scality/cloudserver/pull/1480

**Pull request closed without merging**